src/sphinxfilter.cpp

```diff
@@ -391,12 +391,38 @@ class FilterString_c : public IFilter_Str
 
 protected:
 	SphStringCmp_fn			m_fnStrCmp;
-
-private:
 	CSphFixedVector<BYTE>	m_dVal;
 	bool					m_bEq;
 };
 
+class FilterStringCmp_c : public FilterString_c
+{
+public:
+	FilterStringCmp_c ( ESphCollation eCollation, bool bEq, EStrCmpDir eStrCmpDir )
+		: FilterString_c ( eCollation, bEq )
+		, m_eStrCmpDir ( eStrCmpDir )
+	{}
+
+	bool Eval ( const CSphMatch & tMatch ) const override
+	{
+		auto dStr = tMatch.FetchAttrData ( m_tLocator, m_pBlobPool );
+		int iCmpResult = m_fnStrCmp ( dStr, m_dVal, false );
+
+		switch ( m_eStrCmpDir )
+		{
+			case EStrCmpDir::LT: return m_bEq ? iCmpResult<0 : iCmpResult<=0;
+			case EStrCmpDir::GT: return m_bEq ? iCmpResult>0 : iCmpResult>=0;
```

Comment:
All the tests pass now with the recommended changes implemented. However, this specific part is confusing, because the variable named `m_bEq` suggests that the condition should include "equals" when it is true. However, what `m_bEq` *really* means is the condition is exact - and when `m_bEq` is `false` the condition is inexact (and hence we include the `=`). Hopefully that makes sense. I'm not sure how this can be clarified?